src/posix/platform/radio_spinel.cpp

```diff
@@ -544,46 +545,104 @@ void RadioSpinel::Init(const char *aRadioFile, const char *aRadioConfig)
-                                  &versionMinor));
+exit:
+    SuccessOrDie(error);
+}
 
-        if ((versionMajor != SPINEL_PROTOCOL_VERSION_THREAD_MAJOR) ||
-            (versionMinor != SPINEL_PROTOCOL_VERSION_THREAD_MINOR))
-        {
-            otLogCritPlat("Spinel version mismatch - PosixApp:%d.%d, RCP:%d.%d", SPINEL_PROTOCOL_VERSION_THREAD_MAJOR,
-                          SPINEL_PROTOCOL_VERSION_THREAD_MINOR, versionMajor, versionMinor);
-            exit(OT_EXIT_INCOMPATIBLE_RADIO_SPINEL);
-        }
+otError RadioSpinel::CheckSpinelVersion(void)
+{
+    otError      error = OT_ERROR_NONE;
+    unsigned int versionMajor;
+    unsigned int versionMinor;
+
+    SuccessOrExit(error =
+                      Get(SPINEL_PROP_PROTOCOL_VERSION, (SPINEL_DATATYPE_UINT_PACKED_S SPINEL_DATATYPE_UINT_PACKED_S),
+                          &versionMajor, &versionMinor));
+
+    if ((versionMajor != SPINEL_PROTOCOL_VERSION_THREAD_MAJOR) ||
+        (versionMinor != SPINEL_PROTOCOL_VERSION_THREAD_MINOR))
+    {
+        otLogCritPlat("Spinel version mismatch - PosixApp:%d.%d, RCP:%d.%d", SPINEL_PROTOCOL_VERSION_THREAD_MAJOR,
+                      SPINEL_PROTOCOL_VERSION_THREAD_MINOR, versionMajor, versionMinor);
+        exit(OT_EXIT_INCOMPATIBLE_RADIO_SPINEL);
     }
 
+exit:
+    return error;
+}
+
+otError RadioSpinel::CheckCapabilities(void)
+{
+    otError        error = OT_ERROR_NONE;
+    uint8_t        capsBuffer[kCapsBufferSize];
+    const uint8_t *capsData         = capsBuffer;
+    spinel_size_t  capsLength       = sizeof(capsBuffer);
+    bool           supportsRawRadio = false;
+
+    SuccessOrExit(error = Get(SPINEL_PROP_CAPS, SPINEL_DATATYPE_DATA_S, capsBuffer, &capsLength));
+
+    while (capsLength > 0)
     {
-        const otRadioCaps kRequiredRadioCaps =
-            OT_RADIO_CAPS_ACK_TIMEOUT | OT_RADIO_CAPS_TRANSMIT_RETRIES | OT_RADIO_CAPS_CSMA_BACKOFF;
-        unsigned int caps;
+        unsigned int   capability;
+        spinel_ssize_t unpacked;
+
+        unpacked = spinel_datatype_unpack(capsData, capsLength, SPINEL_DATATYPE_UINT_PACKED_S, &capability);
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_FAILED);
+
+        if (capability == SPINEL_CAP_OPENTHREAD_LOG_METADATA)
+        {
+            mSupportsLogStream = true;
+        }
 
-        SuccessOrExit(error = Get(SPINEL_PROP_RADIO_CAPS, SPINEL_DATATYPE_UINT_PACKED_S, &caps));
-        mRadioCaps = static_cast<otRadioCaps>(caps);
-        if ((mRadioCaps & kRequiredRadioCaps) != kRequiredRadioCaps)
+        if (capability == SPINEL_CAP_MAC_RAW)
         {
-            exit(OT_EXIT_INCOMPATIBLE_RADIO_SPINEL);
+            supportsRawRadio = true;
         }
+
+        capsData += unpacked;
+        capsLength -= static_cast<spinel_size_t>(unpacked);
     }
 
-    mRxRadioFrame.mPsdu = mRxPsdu;
-    mTxRadioFrame.mPsdu = mTxPsdu;
+    if (!supportsRawRadio)
```

Comment:
Do we need check this? After merging mac and link_raw,  will `MAC_RAW` be supported by all OpenThread devices?

Reply:
Even if we combine link-raw and mac, I don't think we would enable link-raw APIs and/or the related spinel properties on all builds. So would be good to check here.

This checks the firmware on NCP/RCP has been built with `OPENTHREAD_RADIO` feature and that radio-only spinel properties which we will use in `radio_spinel` are indeed supported (like `SPINEL_PROP_RADIO_CAPS` or `SPINEL_PROP_MAC_SRC_MATCH_<>` properties).